src/subnet.c

```diff
@@ -161,17 +161,39 @@ subnet_t *lookup_subnet_ipv4(const ipv4_t *address) {
 
 	// Search all subnets for a matching one
 
+	int last_rtt = INT_MAX; // current smallest rtt seen
+
 	for splay_each(subnet_t, p, subnet_tree) {
 		if(!p || p->type != SUBNET_IPV4) {
 			continue;
 		}
 
 		if(!maskcmp(address, &p->net.ipv4.address, p->net.ipv4.prefixlength)) {
-			r = p;
-
-			if(!p->owner || p->owner->status.reachable) {
+			if(!p->owner) {
```

Comment:
The original loop stopped as soon as it hit the smallest reachable subnet that contained the address. The new version does not do this anymore, if there is a small and large subnet that overlap, and the larger subnet has a lower RTT, it will incorrectly choose the larger subnet. Think:

- node A owns Subnet 192.168.1.0/24 and has RTT 20
- node B owns Subnet 192.168.0.0/16 and has RTT 10

With your code, the address 192.168.1.1 will choose B's Subnet. But we only should use RTT to choose between identical subnets. I think the simple solution would be to add this right at the top of the body of the `if`-statement:

```
if (r && r->net.ipv4.prefixlength != p->net.ipv4.prefixlength) {
    break;
}
```